packages/components/src/TooltipTrigger/TooltipTrigger.component.js

```diff
@@ -22,17 +24,75 @@ const props = {
 	<Icon name="my-icon" />
 </TooltipTrigger>
  */
-function TooltipTrigger(props) {
-	const tooltip = (<Tooltip className={getTooltipClass()} id={uuid.v4()}>{props.label}</Tooltip>);
-	return (
-		<OverlayTrigger
-			placement={props.tooltipPlacement}
-			overlay={tooltip}
-			delayShow={400}
-		>
-			{props.children}
-		</OverlayTrigger>
-	);
+class TooltipTrigger extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onMouseOver = this.onMouseOver.bind(this);
+		this.onMouseOut = this.onMouseOut.bind(this);
+		this.state = {
+			hovered: false,
+			id: uuid.v4(),
+		};
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
```

Comment:
@sdwvit This make the code harder to read. Here it is clear, if the conditions were more complicated yes that's a good point to externalize it. But here it is simpler to keep this easy checks IMO